lib/json_api/json_api.rb

```diff
@@ -38,14 +40,28 @@ def listen_for_user_input(game)
           game = @game_deserializer.deserialize(request_body)
           game = game.make_move(self) unless game.game_over?
           json @game_serializer.serialize(game)
-        rescue
-          json "error" => "invalid game or players"
+        rescue DeserializationError, Game::InvalidPositionError => message
+          json "error" => message
+        rescue Exception
+          json "error" => "an unknown error occured"
         end
       end
     end
 
     private
 
+    def missing_move?(request_body)
+      !request_body.has_key?('move') ||
```

Comment:
Alternatively, use `.all?`

Reply:
Couldn't figure out how that would work. Inverted the boolean logic though.